src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/CISShadowMonitor.java

```diff
@@ -226,6 +228,14 @@ private synchronized void processCISShadow(String version, Map<String, Object> d
                         .log("Retry workflow for getting connectivity info interrupted");
                 Thread.currentThread().interrupt();
                 return;
+            } catch (ValidationException | ResourceNotFoundException e) {
+                LOGGER.atWarn().cause(e).log("Connectivity info doesn't exist");
+                try {
+                    updateCISShadowReportedState(desiredState);
```

Comment:
this is actually publishing to the cloud. Why would we publish anything if our request was invalid or the shadow wasn't found.

Reply:
yep, was considering those situations to be non-retyable, so updating as a means of signaling that we are done processing that particular shadow version

just my initial thinking, not hard committed to this if you think otherwise

Reply:
add a comment in the code to explain this.

Reply:
added